README.md

```diff
@@ -4,204 +4,430 @@ Daniel Ehrenberg, Igalia. Stage 3
+5n / 2n
+// ↪ 2n
 
-### Don't break math
+Number(151851850485185185047n)
+// ↪ 151851850485185200000
 
-The semantics of all operators should ideally be based on some mathematical first principles, and certainly be well-defined, rather than excessively exposing implementation artifacts. `/` and `%` round towards 0, this is to match well-established computer conventions; aside from that, all operators have clean, mathematical definitions which don't appeal to the implementation shape.
+```
 
-### Don't break asm.js
+### Other Exceptions
 
-Although this proposal introduces operator overloading, it throws in any of the cases that asm.js depends on for setting up type checking. asm.js relies on a few identities:
-- Unary `+` followed by an expression is always either a Number, or results in throwing. For this reason, unfortunately, `+` on a BigInt needs to throw, rather than being symmetrical with `+` on Number: Otherwise, previously "type-declared" asm.js code would now be polymorphic.
-- `|0` always returns a Number in int32 range, or throws. This proposal maintains that, as it would throw on a BigInt for being a mixed operand type.
-- `Math.fround` always returns a Number in float32 range, or throws. This proposal would throw if `Math.fround` is called with a BigInt, preserving the property.
+Attempting to convert a fractional value to a `BigInt` throws an exception both when the value is represented as an `Number` and a `String`.
+
+```js
+BigInt(1.5)
+// ↪ RangeError: The number 1.5 is not a safe integer and thus cannot be converted to a BigInt
+
+BigInt('1.5')
+// ↪ SyntaxError: Cannot convert 1.5 to a BigInt
+
+```
+
+`Math.round`, `Math.fround` and `|` throw exceptions when called with a `BigInt` since these functions are defined as returning a `Number`.
+
+```js
+
+Math.round(1n)
+// ↪ TypeError: Cannot convert a BigInt value to a number
+
+Math.fround(1n)
+// ↪ TypeError: Cannot convert a BigInt value to a number
+
+1n|0
+// ↪ TypeError: Cannot mix BigInt and other types, use explicit conversions
+
+```
+
+### Usage Recommendation
+
+It is recommended to continue using `Number` for code which will only encounter values under 2<sup>53</sup>.
+
+Reserve `BigInt` for cases where large values are expected.
+
+## About the Proposal
+
+### Motivation: Why Do We Need Such Big Numbers?
+
+There are a number of cases in JavaScript coding where integers larger than 2<sup>53</sup> come up — both instances where signed or unsigned 64-bit integers are needed and times where we may want integers even larger than 64-bits.
+
+#### 64-bit Use Cases
+
+Often, other systems with which Javascript interacts provides data as 64-bit integers, which lose precision when coerced to Javascript numbers.
+
+These might come when reading certain machine registers or wire protocols or using protobufs or JSON documents that have GUIDs generated by 64-bit systems in them — including things like credit card or account numbers.
```

Comment:
Note: Since we can't make changes to the JSON format, it actually triggers an error to serialize BigInt directly. But you can use libraries like https://github.com/kanongil/granola to serialize and serialize BigInt and other JS datatypes to JSON.